app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,39 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:user_id])
+    @posts = @user.posts
```

Comment:
Please ensure that you solve N+1 problem when fetching all posts and their comments for a user according to the project requirement.
Example in the posts#index actions you should use @posts = @user.posts.includes(:comments) instead of @posts = @user.posts